File: 백준/Silver/15649. N과 M （1）/N과 M （1）.swift
let input = readLine()!.split(separator: " ").map{Int($0)!}
let N = input[0]
let M = input[1]

var nums:[Int] = Array(repeating: 0, count: M)
var ch:[Bool] = Array(repeating: false, count: N)

getResult(0)

func getResult(_ idx: Int) -> Void {
    if(idx == M) {
        var str = ""
        for i in 0..<M { str += "\(nums[i]) " }
        print(str)
        return
    }
    for i in 0..<N {
        if(!ch[i]) {
            nums[idx] = i+1
            ch[i] = true
            getResult(idx+1)
            ch[i] = false
        }
    }
}

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    static int[] nums;
    static boolean[] ch;
    static int N, M;
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        M = sc.nextInt();

        nums = new int[M];
        ch = new boolean[N];

        getResult(0);
    }

    static void getResult(int idx) {
        if(idx == M) {
            String str = "";
            for(int i=0; i<M; i++) str += nums[i] + " ";
            System.out.println(str);
            return;
        }
        for(int i=0; i<N; i++) {
            if(!ch[i]) {
                nums[idx] = i+1;
                ch[i] = true;
                getResult(idx+1);
                ch[i] = false;
            }
        }
    }
}